src/nix/cat.cc

```diff
@@ -6,21 +6,21 @@ using namespace nix;
 
 struct MixCat : virtual Args
 {
-    std::string path;
```

Comment:
This seems to duplicate `std::string path` unnecessarily. (Also we should probably change it to `CanonPath`.)

Reply:
@edolstra I duplicated it because the two paths are parsed differently. With one, we parse a store path (and remainder). With the other, we directly convert to a `CanonPath`.